### ISSUE TITLE
Support the specification of an application instance for the translation tool

### ISSUE THREAD

**Issue by elfring** [symfony/symfony#40577]
I hope that [the provided translation tool will become better usable](https://github.com/symfony/symfony/issues/40570 "Export analysis results according to the translation process").

:thinking: I imagine that another development challenge is to distinguish the storage for translated data according to application containers.
Would you like to [support additional command parameters accordingly](https://symfony.com/doc/current/translation.html#extracting-translation-contents-and-updating-catalogs-automatically "Extracting translation contents and updating catalogs automatically")?

**Comment by xabbuh** [comment]
> I imagine that another development challenge is to distinguish the storage for translated data according to application containers.

Can you explain a bit more what you have in mind here? Currently, I feel like I don't have any idea what that would mean in practice.

**Comment by elfring** [comment]
The support for a command like “`php bin/console translation:update` …” can be configured in a few directories so that data can be extracted for desired translations.
:thinking: I imagine then that this translation tool will be usable for some applications (or even containers) which are stored in subdirectories.
Thus the tool needs probably to know to which application containers translatable data belong.

**Comment by xabbuh** [comment]
I still have no idea what you are asking for. Can you please share a use case that you have right now for which there is no solution and what exact solution you have in mind?

**Comment by elfring** [comment]
I am trying to clarify the distinction for input and output directories according to the safe application of a command like “`php bin/console translation:update` …”.

I assume that you can be used to directory structures for application containers.
Do I expect too much here?

**Comment by xabbuh** [comment]
> Do I expect too much here?

I don't know because honestly I still simply do not understand what you would like to see changed.

**Comment by dbrumann** [comment]
The debug tooling will tell you when you use a translation that is not yet defined in any of the configured paths and then set those missing translation keys up in your default path.

You can already collect translations from different directories. You can reuse translations, e.g. by providing them in a bundle, and then override them in your application giving you some kind of hierarchy. You can also pull in translations from different locations. You can only update missing translations in your default translation path, to make sure they have higher priority than any new directory you might pull in, that could overwrite them. That does not stop you from copying them over, e.g. into a bundle or a separate path that might be used by more projects. Adding functionality for updating translations in other locations than your main project to me seems out of scope for the update command.

If I understand correctly, you want translations to be updated in other places than your project. There are certain limitations. For example the order in which the directories are search in, the translations might overwrite each other. That might be desirable. For example Symfony Validation provides translations for validation messages, that can be overwritten in your project's translations. The same goes for security related messages.

In short, as far as I can tell everything you propose is already possible. Having the update command write to a different directory does not make much sense in the context of the app the command is running in. If I misunderstood or jumped to a wrong conclusion, please provide a concrete example. Maybe you can point to an existing solution in a different programming language or you can draw up a quick diagram to show what your solution would do differently than what the component currently does.

**Comment by elfring** [comment]
>Adding functionality for updating translations in other locations than your main project to me seems out of scope for the update command.

I came along special software development needs.

Such a “project” can consist of some components.
They can be also application containers in subdirectories, can't they?


>If I understand correctly, you want translations to be updated in other places than your project.

Probably not so far. — Our imaginations are evolving according to possible directory structures.

Example:
my_root
+ my_container-X
+ my_container-Y
+ …


>Having the update command write to a different directory does not make much sense in the context of the app the command is running in.

I got other development views.

**Comment by dbrumann** [comment]
My assumption is, that what you call containers are pretty much isolated services that can be reused between applications. In that case they are probably not managed by your project itself, meaning the command should not write in them, because that could affect other applications using the component or at least will require them to be "notified" as well, i.e. by means of having a new release.

The goal of the TranslationUpdateCommand is to give a better developer experience for your Symfony application, which is why it is provided via the framework-bundle (an extension point specifically for Symfony apps). If you want to write a command that is providing similar functions, but for other projects or containers, than in my opinion this is a separate tool that should not be part of Symfony itself and instead is something you could write as a dedicated console command for your needs and maybe build on the existing resources.

Symfony is an open source project that can not reasonably maintain all things around Symfony. Instead the team focuses on making Symfony itself better and providing helpful tools around somewhat common use cases in a typical Symfony app. To me it seems you are looking for something that is outside that scope and - much like 3rd party bundles and apps built on Symfony - it's own thing.

**Comment by elfring** [comment]
>In that case they are probably not managed by your project itself,

Your view does not match with my development views at the moment.


>meaning the command should not write in them, …

It seems that your imaginations are too limited for my concrete use case so far.


>To me it seems you are looking for something that is outside that scope …

* Do other software developers care more for consistent translations according to contents from application containers?
* Can any interests be increased by donating additional development resources?


How much are you used to the command “`translation:update`” as a general console application for related systems?

**Comment by xabbuh** [comment]
> How much are you used to the command “`translation:update`” as a general console application for related systems?

The command provided by the FrameworkBundle is not meant to be used across different applications. I agree with @dbrumann that a tool like that should indeed better be part of a separate package. So let's close here. Thank you for understanding.
